### PR TITLE
Storyshots-puppeteer: Add afterScreenshot handler

### PR DESCRIPTION
**Pull request by svewag**
In order to be able to work with the just created image, I added a `afterScreenshot` handler.

### CODE REVIEW

File: addons/storyshots/storyshots-puppeteer/src/index.js [hunk 1]
```diff
@@ -65,6 +67,7 @@ export const imageSnapshot = (customConfig = {}) => {
       await page.goto(url, getGotoOptions({ context, url }));
       await beforeScreenshot(page, { context, url });
       image = await page.screenshot(getScreenshotOptions({ context, url }));
+      await beforeScreenshot({ image, context });
```
Comment (shilman) [inline addons/storyshots/storyshots-puppeteer/src/index.js]: shouldn't that be `afterScreenshot`? have you tested this?

Comment (svewag) [reply addons/storyshots/storyshots-puppeteer/src/index.js]: oh boy 🤦‍♂

**Comment by svewag** [comment]
Is there anything I can do so that let the PR will be accepted? Me and my team, we are really looking forward to use this handler in our project.

I know you are very busy, but maybe there is a chance. If there is anything I can do, please let me know.

Thank you so far for your awesome work 🙏

**Comment by shilman** [comment]
@svewag can you update from `next` & resolve the conflicts & update yarn.lock?

**Comment by shilman** [comment]
@svewag typescript error?

```
lerna ERR! yarn run prepare exited 2 in '@storybook/addon-storyshots-puppeteer'
lerna ERR! yarn run prepare stdout:
$ node ../../../scripts/prepare.js
src/imageSnapshot.ts(21,3): error TS2322: Type '{ storybookUrl: string; chromeExecutablePath: undefined; getMatchOptions: () => undefined; getScreenshotOptions: () => { readonly fullPage: true; readonly encoding: "base64"; }; beforeScreenshot: () => undefined; afterScreenshot: () => undefined; getGotoOptions: () => undefined; customizePage: () => Promise<...>; ge...' is not assignable to type 'ImageSnapshotConfig'.
  Object literal may only specify known properties, and 'afterScreenshot' does not exist in type 'ImageSnapshotConfig'.
src/imageSnapshot.ts(34,5): error TS2525: Initializer provides no value for this binding element and the binding element has no default value.
```

**Comment by svewag** [comment]
I think now everything is fine

**Comment by svewag** [comment]
so, what's next?

**Review by ndelangen — APPROVED** [review]

**Pull request merged**